test/built-ins/Promise/prototype/finally/this-value-then-poisoned.js

```diff
@@ -0,0 +1,20 @@
+// Copyright (C) 2017 Jordan Harband. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+author: Jordan Harband
+description: >
+  Promise.prototype.finally called with a `this` value whose `then` property is
+  an accessor property that returns an abrupt completion
+esid: sec-promise.prototype.finally
+features: [Promise, Object.defineProperty, Promise.prototype.finally]
+---*/
+
+var poisonedThen = Object.defineProperty(new Promise(function () {}), 'then', {
+  get: function() {
+    throw new Test262Error();
+  }
+});
+
+assert.throws(Test262Error, function() {
+  Promise.prototype.finally.call(poisonedThen);
```

Comment:
this would like calling `poisonedThen.finally()`, right? We can add a new `assert.throws` in this file.